File: back/Dto/ProductCreateOrEditInputDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace back.Dto
{
    public class ProductCreateOrEditInputDto
    {
        public long? ProductId { get; set; }
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public string UnitOfMeasure { get; set; }
        public int UnitPrice { get; set; }
        public int UnitOfCurrency { get; set; }
        public string Status { get; set; }
        public DateTime Date { get; set; }
        public long CategoryId { get; set; }
    }
}

File: front/src/app/_Dto/ProductCreateOrEditInputDto.ts
export class ProductCreateOrEditInputDto {
    productId: number;
    productName: string;
    quantity: number;
    unitOfMeasure: string;
    unitPrice: number;
    unitOfCurrency: string;
    status: string;
    date: string;
    categoryId: number;
    constructor() {
            this.productId = 0;
            this.productName = "";
            this.quantity = 0;
            this.unitOfMeasure = "";
            this.unitPrice = 0;
            this.unitOfCurrency = "";
            this.status = "";
            this.date = "";
            this.categoryId = 0;
    }
}